File: python/raindrops/raindrops.py
def convert(number: int) -> str:
    sounds = {
        3: "Pling",
        5: "Plang",
        7: "Plong"
    }

    results = (
        sound for divisor, sound in sounds.items() if number % divisor == 0
    )

    return "".join(results) or str(number)

File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(number)
    sounds = {
      3 => "Pling",
      5 => "Plang",
      7 => "Plong"
    }

    result = sounds
               .select { |k, _| number % k == 0 }
               .values
               .join

    result.empty? ? number.to_s : result
  end
end